elastic/kibana
# [Console] Settings view enhancements

## Summary

This PR combines the settings and variables modals into a single top level `config` view. Besides that it brings a few QOL enhancements to both flows:

* Settings editor:
  * The form has a new layout grouping settings accordingly to its feature
  * All changes take effect right away without having to `submit` the form
* Variables editor:
  * The form has a new layout that makes it easier to understand how the variables are used
  * When deleting a variable, a confirm modal is shown
  * Editing a variable happens inline directly in the variables table
  * All changes take effect right away without having to `submit` the form 

##### How to test

* Start up es and kibana
* Navigate to `Devtools`
* Verify that:
  * Verify settings can be modified accordingly
  * Verify variables can be edited and that they work
  * Only one variable can be edited at the same time


https://github.com/user-attachments/assets/51b4e50d-092e-4de6-8516-086369209a95



<details>
<summary>UI screenshots</summary>

![Screenshot 2024-08-24 at 07 27 33](https://github.com/user-attachments/assets/b798843d-9878-4c01-9ce5-cdd5a9265115)
![Screenshot 2024-08-24 at 07 27 50](https://github.com/user-attachments/assets/a7f71fa7-feb3-4bcc-97cf-1492d9a59f32)
![Screenshot 2024-08-24 at 07 27 45](https://github.com/user-attachments/assets/62bb097d-a25a-4bf0-8dcb-86767b06c428)
![Screenshot 2024-08-24 at 07 27 39](https://github.com/user-attachments/assets/1daeb618-b487-4774-b3a9-78a17cffa8b0)
![Screenshot 2024-08-26 at 16 01 48](https://github.com/user-attachments/assets/b112ab10-5d69-48c5-aa17-62e156cefce0)


</details>
